File: test files/java/junior_medio.java
import java.util.ArrayList;
import java.util.List;

public class JuniorMedio {
    enum TaskStatus {
        TODO, DOING, DONE
    }

    static class Task {
        int id;
        String title;
        String owner;
        TaskStatus status;

        Task(int id, String title, String owner, TaskStatus status) {
            this.id = id;
            this.title = title;
            this.owner = owner;
            this.status = status;
        }
    }

    public static void main(String[] args) {
        List<Task> tasks = new ArrayList<>();
        tasks.add(new Task(1, "Criar backlog", "ana", TaskStatus.TODO));
        tasks.add(new Task(2, "Revisar ticket", null, TaskStatus.DOING));

        Task task = findTask(tasks, 2);
        if (task != null) {
            assign(task, "bruno");
            moveToDone(task);
        }
    }

    static Task findTask(List<Task> tasks, int id) {
        for (Task task : tasks) {
            if (task.id == id) {
                return task;
            }
        }
        return null;
    }

    static void assign(Task task, String owner) {
        task.owner = owner;
    }

    static boolean moveToDone(Task task) {
        if (task.owner == null || task.owner.isBlank()) {
            return false;
        }

        task.status = TaskStatus.DONE;
        return true;
    }
}

File: test files/typescript/junior_medio.ts
type TaskStatus = 'todo' | 'doing' | 'done';

type Task = {
  id: number;
  title: string;
  owner: string | null;
  status: TaskStatus;
};

const tasks: Task[] = [
  { id: 1, title: 'Criar backlog', owner: 'ana', status: 'todo' },
  { id: 2, title: 'Revisar ticket', owner: null, status: 'doing' },
];

function getTaskById(id: number): Task | null {
  return tasks.find((task) => task.id === id) ?? null;
}

function moveTask(id: number, status: TaskStatus): { ok: boolean; message: string } {
  const task = getTaskById(id);

  if (!task) {
    return { ok: false, message: 'Task nao encontrada' };
  }

  if (status === 'done' && !task.owner) {
    return { ok: false, message: 'Task precisa ter responsavel antes de concluir' };
  }

  task.status = status;
  return { ok: true, message: `Task ${task.title} movida para ${status}` };
}

function assignTask(id: number, owner: string): Task | null {
  const task = getTaskById(id);

  if (!task) {
    return null;
  }

  task.owner = owner.trim();
  return task;
}

assignTask(2, 'bruno');
console.log(moveTask(2, 'done'));
